cmd/kube-scheduler/app/options/options.go

```diff
@@ -122,11 +122,11 @@ func splitHostIntPort(s string) (string, int, error) {
 	if err != nil {
 		return "", 0, err
 	}
-	portInt, err := strconv.Atoi(port)
+	portInt, err := netutils.ParsePort(port, false)
```

Comment:
Using this case to open the discussion.  Not specific to this case.

On one hand, Atoi() will happily return 0, so for strict compatibility we should allow 0.

On the other hand, it's not clear that this would actually be right, given the callers. of it.

We should probably add a comment here that this presumes that 0 is not allowed as a port value.  I expect I will find other places in similar form

Reply:
Agreed. Do you think this warrants a comment everywhere it's called?

Reply:
@thockin the 0 value is used to obtain a free random port in the net.Listen() function, don't know if this is the case here but I guess it's possible that other places expect a 0 for the port number

>  If the port in the address parameter is empty or "0", as in "127.0.0.1:" or "[::1]:0", a port number is automatically chosen.
https://golang.org/pkg/net/#Listen

Reply:
@aojea Correct.  My point was that the existing (previous) code uses Atoi() which allows 0, so not allowing 0 is a behavioral change.  But, I don't think we WANT to allow 0 here (given the callers and scope of this function), so this could actually be considered a bugfix.

Reply:
should we review this decision for the rest of the changes in this PR or is safe to consider that 0 is not allowed?

Reply:
Pinging this thread - I have not allowed zero for the most part except for in flags.go. PTAL